### DISCUSSION TITLE
Documentação e padrões de componentes / design

### DISCUSSION THREAD

**Discussion by guilhermethalesmereo** [frontendbr/forum#1486] [Discussão]
Pessoal, gostaria de saber como e onde vocês fazem uma documentação de tudo referente ao design de componentes e explicações de quando usar cada elemento dentro do produto em que vocês atuam.

Estou com um problema hoje onde trabalho, pois existe uma página na internet onde existem alguns componentes que o sistema usa, porém está muito desatualizado. Uma designer fez um documento em PDF onde falava sobre os componentes, adicionando alguns novos também, mas não sei se documento escrito assim seria o ideal.

Outra coisa também que gera bastante dor de cabeça é desenvolver componentes "fora do padrão". Pois como os documentos são antigos, quando desenvolvemos um módulo novo como estamos fazendo agora, em alguns casos específicos, o comportamento das novas telas não é igual ao de telas de módulos antigos já do sistema, o que acaba gerando retornos por estar "fora do padrão".

Se puderem me indicar algumas soluções que vocês utilizam para fazer essas documentações de estilo e componentes de forma fácil... Desde já agradeço!

**Comment by Juuhz** [comment]
Fala Dev,

Acho que o Story Book pode te ajudar, da uma olhada:
https://storybook.js.org/

Espero ter conseguido ajudado, abraços.

**Comment by ninetails** [comment]
Sobre documentação, também sugiro o Storybook (acima).

Já em relação ao desenvolver componentes fora do padrão, isso é problema de desenvolvimento. Claro, uma boa documentação vai ajudar, mas sugiro também pensar em alguma solução como **Testes de Regressão Visual** e um bom _Git workflow_ com **Code Review**.
